airflow/sensors/filesystem.py

```diff
@@ -38,23 +38,27 @@ class FileSensor(BaseSensorOperator):
     :param filepath: File or folder name (relative to
         the base path set within the connection), can be a glob.
     :type filepath: str
+    :param recursive: when set True, enables recursive directory matching behavior of
+        `**` in glob filepath parameter. Defaults to False.
```

Comment:
An example would be nice too

Reply:
Do you mean an example of invocation like below? Or an example scenario where this would be used like #16725?
```
FileSensor(
            task_id='test',
            filepath=temp_dir + "/**",
            fs_conn_id='fs_default',
            dag=self.dag,
            timeout=0,
            poke_interval=1,
            recursive=True,
        )
```
Also, would this example go to the same place? The docstring for the class?

Reply:
An example in `airflow/example_dags` it was missing for filesystem_sensor, but it would be nice to add one since we are changing it.

Reply:
```suggestion
    :param recursive: when set to ``True``, enables recursive directory matching behavior of
        ``**`` in glob filepath parameter. Defaults to ``False``.
```

Reply:
Feel free to do that in a follow-up PR to add an example DAG